apps/application-system/api/src/app/modules/application/lifecycle/application-lifecycle.service.ts

```diff
@@ -80,13 +74,17 @@ export class ApplicationLifeCycleService {
   }
 
   private async pruneApplicationCharge() {
-    for (const prune of this.processingApplications) {
+    for (const prune of this.processingApplications.filter(
+      (application) => !!application.application.paymentId,
+    )) {
       try {
         await this.applicationChargeService.deleteCharge(prune.application)
-      } catch (error) {
-        const chargeId = this.applicationChargeService.getChargeId(
-          prune.application,
+        console.log(
+          'delete charge from application Id : ',
+          prune.application.id,
```

Comment:
Console log?
Should this be using logger or is this perhaps running in a client context?